### ISSUE TITLE
Dashlane password manager breaks redraw

### ISSUE THREAD

**Issue by tropperstyle** [leeoniya/domvm#128]
<img width="824" alt="screen shot 2017-04-20 at 5 06 28 pm" src="https://cloud.githubusercontent.com/assets/28314/25257409/afdf9b48-25ec-11e7-810f-a3e7db274ede.png"> 

I'm currently running into this `node is  undefined` error. If I step back in the call stack to syncChildren I see that `lftSib` is span tag that is injected by the Dashlane plugin for Firefox. Here is a screenshot of the element:

<img width="272" alt="screen shot 2017-04-20 at 5 11 17 pm" src="https://cloud.githubusercontent.com/assets/28314/25257499/4289b8b6-25ed-11e7-92d5-811ee089d925.png">

I have no idea if this is a simple null check on `lftSib._node`, but I am guessing domvm needs to ignore elements it doesn't know about. I don't know much about the internals of a virtual dom engine, so I apologize for the lack of a PR attempt, but please let me know if you need more information or an isolated use case.

I know it must suck to account for 3rd party actors. I am porting over a complex mithril app, and this is the only thing that is tripping me up so far. Everything else feels so natural. It is amazing at how powerful domvm is with such a small amount of code living in the application space. I was very disappointed after much excitement about the mithril 1.0 rewrite, but glad to have found this project.

Thank you!

**Comment by leeoniya** [comment]
hey @tropperstyle 

glad you're enjoying domvm. if you use it enough and wanna help out, i need help with docs!

> I have no idea if this is a simple null check on `lftSib._node`, but I am guessing domvm needs to ignore elements it doesn't know about.

yeah this will probably work, and would also have to be done for `rgtSib._node` lower in the loop. as well as some other funcs in that syncChildren file which expect both _node and _node.idx to be present.

my concern would be the perf impact here since it's a hot function, but there's actually a pretty juicy case here for taking a small perf hit to accomodate interop with eg jquery plugins or vanilla js libs that insert stuff into existing/predefined structures. i'll do some benchmaking tonight to see how much perf this will cost.

**Comment by leeoniya** [comment]
ok, i added some handling but took a slightly different route.

rather than null-testing in several places, i attach fake `el._node` objects which contain just the necessary bits for the reconciler to work as it does normally `{el, parent, idx}`. the idx is fractionally incremented (or decremented) by multiples of 0.01 from the prior real index, allowing the selection sorting fallback to work without special handling/skipping...hopefully. i added a very simple test for this which appears to work and didn't see any perf drop in the reconciler (tested via 0% dbmonster mutations) [1]

let me know if this patch fixes your case.

[1] https://rawgit.com/leeoniya/domvm/2.x-dev/demos/bench/dbmonster/index.html

**Comment by tropperstyle** [comment]
TypeError: lftNode is undefined

```
{ lsNode = lftSib._node = {parent: node, idx: lftNode.idx + (++lftFaux * .01)}; }
```

**Comment by leeoniya** [comment]
i added another test that discovered obvious issues with inserting faux nodes and have pushed a commit that implements the simpler skipping behavior. please retry.

as mentioned in the commit, i have not added the skipping to some of the helper reconciler functions that are meant to handle more complex reordering cases, so it's still not bulletproof, but i'd rather not bloat the codebase for rare cases if not actually encountered in real apps.

**Comment by leeoniya** [comment]
if this whole thing becomes too hairy, then i'll probably back it all out and we can figure out a app-space solution.

**Comment by tropperstyle** [comment]
> i imagine that most (all?) vdom libs will not tolerate third party code trashing their meticulously-tracked dom and will result in a Pretty Bad Time (tm) - undefined/random behavior all over the place.

I would assume the same, but for what it is worth, I have used Dashlane for a couple years now and this is the first time I needed to disable the browser plugin to prevent a page from breaking.  I also did not have this problem with mithril 0.2.x or 1.x. I might try and dig deeper into when Dashlane decides to inject those elements. Maybe it is not injecting anything at all with the way mithril manipulates the dom.

In any case, I am no longer running into an error with your latest commit, so it seems to have done the trick.

Thanks for the quick fix.
